ow_dynamic_terrain/src/DynamicTerrainVisual.h

```diff
@@ -0,0 +1,66 @@
+// The Notices and Disclaimers for Ocean Worlds Autonomy Testbed for Exploration
+// Research and Simulation can be found in README.md in the root directory of
+// this repository.
+
+#include "DynamicTerrainBase.h"
+#include "TerrainModifier.h"
+#include <gazebo/rendering/RenderingIface.hh>
+
+namespace ow_dynamic_terrain
+{
+class DynamicTerrainVisual : public gazebo::VisualPlugin, public DynamicTerrainBase
+{
+public:
+  DynamicTerrainVisual();
+
```

Comment:
Though I don't think it's in our coding guidelines, I like to see all classes either define, `delete`, or `default` the copy constructor, destructor, and assignment operator.

Reply:
I'll propose it in a future meeting.  It simply makes explicit the intention.  Without anything you get `default` but it's good to know that was intended.

Reply:
I think it's a good practice when you have vital code in a class's custom constructor, but not in a default constructor, such that if the default constructor were to be used you would more then certainly wind up with an error.

That said, we are talking about a Gazebo plugin here, so the only thing that will ever construct it is Gazebo code that neither us nor our users should ever touch.

Reply:
Actually I just talked myself into feeling that deleting unused constructors is good even for a Gazebo plugin like this. As unlikely as it may be, if Gazebo had an update where it now wants to pass some argument to its plugin constructors, or it for some reason has to copy them now, you would know what the problem is immediately from a compile error, but if you had not deleted unwanted constructors you may have to painstakingly find out by observing that your class is now erring and working backwards to find out why.